src/lib/bootloader/stage1.rb

```diff
@@ -139,9 +145,11 @@ def extended_boot_partitions_names
     end
 
     def extended_boot_partition?
-      return false if boot_partition_names == extended_boot_partitions_names
+      names = extended_boot_partitions_names
+      return false if names.empty?
```

Comment:
np: why not merging this line and the next one?

Reply:
I like to have non related checks separated for readability